sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -461,68 +462,269 @@ case class FindInSet(left: Expression, right: Expression) extends BinaryExpressi
 }
 
 /**
- * A function that trim the spaces from both ends for the specified string.
- */
+ * A function that takes a character string, removes the leading and/or trailing characters matching with the characters
+ * in the trim string, returns the new string. If LEADING/TRAILING/BOTH and trimStr keywords are not specified, it
+ * defaults to remove space character from both ends.
+ * trimStr: A character string to be trimmed from the source string, if it has multiple characters, the function
+ * searches for each character in the source string, removes the characters from the source string until it
+ * encounters the first non-match character.
+ * LEADING: removes any characters from the left end of the source string that matches characters in the trim string.
+ * TRAILING: removes any characters from the right end of the source string that matches characters in the trim string.
+ * BOTH: removes any characters from both ends of the source string that matches characters in the trim string.
+  */
 @ExpressionDescription(
-  usage = "_FUNC_(str) - Removes the leading and trailing space characters from `str`.",
+  usage = """
+    _FUNC_(str) - Removes the leading and trailing space characters from `str`.
+    _FUNC_(BOTH trimStr FROM str) - Remove the leading and trailing trimString from `str`
+    _FUNC_(LEADING trimStr FROM str) - Remove the leading trimString from `str`
+    _FUNC_(TRAILING trimStr FROM str) - Remove the trailing trimString from `str`
+  """,
   extended = """
+    Arguments:
+      str - a string expression
+      trimString - the trim string
+      BOTH, FROM - these are keyword to specify for trim string from both ends of the string
+      LEADING, FROM - these are keyword to specify for trim string from left end of the string
+      TRAILING, FROM - these are keyword to specify for trim string from right end of the string
     Examples:
       > SELECT _FUNC_('    SparkSQL   ');
        SparkSQL
+      > SELECT _FUNC_(BOTH 'SL' FROM 'SSparkSQLS');
+       parkSQ
+      > SELECT _FUNC_(LEADING 'paS' FROM 'SSparkSQLS');
+       rkSQLS
+      > SELECT _FUNC_(TRAILING 'SLQ' FROM 'SSparkSQLS');
+       SSparkS
```

Comment:
yah, SSPark. But I will remove the Examples based on your previous comments about the usage.